examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/AppPlatformService.java

```diff
@@ -98,17 +118,39 @@ public void setActivity(Activity activity) {
   }
 
   public void addContentApp(ContentApp app) {
-    app.setEndpointId(
-        mAppPlatform.addContentApp(
-            app.getVendorName(),
-            app.getVendorId(),
-            app.getAppName(),
-            app.getProductId(),
-            "1.0",
-            new ContentAppEndpointManagerImpl(context)));
+    int retEndpointId = -1;
+    int desiredEndpointId = discoveredEndpoints.getInt(app.getAppName(), -1);
+    if (desiredEndpointId > 0) {
+      retEndpointId =
+          mAppPlatform.addContentAppAtEndpoint(
+              app.getVendorName(),
+              app.getVendorId(),
+              app.getAppName(),
+              app.getProductId(),
+              "1.0",
+              desiredEndpointId,
+              new ContentAppEndpointManagerImpl(context));
+    } else {
+      retEndpointId =
+          mAppPlatform.addContentApp(
+              app.getVendorName(),
+              app.getVendorId(),
+              app.getAppName(),
+              app.getProductId(),
+              "1.0",
+              new ContentAppEndpointManagerImpl(context));
+    }
+    if (retEndpointId > 0) {
+      app.setEndpointId(retEndpointId);
+      discoveredEndpoints.edit().putInt(app.getAppName(), app.getEndpointId()).apply();
+    }
   }
 
-  public int removeContentApp(int endpointID) {
-    return mAppPlatform.removeContentApp(endpointID);
+  public int removeContentApp(int endpointID, String appName) {
+    int retEndpointId = mAppPlatform.removeContentApp(endpointID);
+    if (endpointID == retEndpointId) {
```

Comment:
what does it mean for endpointID to be different from retEndpointId? is this checking just for 0/-1 for "not found" or is there more logic to it. 

Why does 'removeContentApp(endpointID)' return an endpointid instead of a bool or error?

Reply:
Based on the [TODO](https://github.com/project-chip/connectedhomeip/pull/22354/files/5534efb983342385556df870e15c9a06c1fc68d6#diff-78f855fb33f255484885e18b9af8c228261872b3273b6c041d19d8a11f60537dR77) this code will be removed as we will not be removing endpoints once added. 

The [ContentAppPlatform](https://github.com/project-chip/connectedhomeip/blob/master/src/app/app-platform/ContentAppPlatform.h#L100) in the SDK returns the removed endpoints ID. That gets passed back through all layers to here. Not sure if we can update that. But that will not be relevant for this particular example app anymore after the above mentioned change.